### PR TITLE
[vcpkg baseline][vcpkg-ci-msys2] Update msys2

### PR DESCRIPTION
**Pull request by dg0yt**
`msys2-runtime` 3.4.10 (maybe resolving runtime regressions (#35989).)  
`auoconf2.72`  
`mingw-w64-x86_64-gettext-runtime` replacing `mingw-w64-x86_64-gettext`. Fixes vcpkg-ci-msys2 baseline regression:
~~~
CMake Error at scripts/test_ports/vcpkg-ci-msys2/portfile.cmake:310 (message):
  The following msys2 packages are no longer in the database:
  mingw-w64-x86_64-gettext
~~~

Changes how `vcpkg_find_acquire_program(PKGCONFIG)` and `vcpkg_find_fortran()` request msys packages from `DIRECT_PACKAGES` to fully declared packages with explicit `DEPS` and `PROVIDES` as in `vcpkg_acquire_msys()`.
This change makes it possible to detect changed dependencies in `vcpkg-ci-msys2` update checks. (Current case: zstd.)
It also enables reuse of general declarations from `vcpkg_acquire_msys` in specialized cases (e.g. x64 gcc libs for `vpckg_find_fortran`). 
Last not least, using the new `Z_DECLARE_EXTRA_PACKAGES_COMMAND` simplifies the implementation of `vcpkg-ci-msys2` and makes it easy to manage complete msys updates from any platform.

### CODE REVIEW

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by MonicaLiu0311** [comment]
Resolved the error with Pipelines: https://dev.azure.com/vcpkg/public/_build/results?buildId=99214&view=results

File: scripts/cmake/vcpkg_find_fortran.cmake [hunk 1]
```diff
@@ -132,3 +64,149 @@ function(vcpkg_find_fortran out_var)
     endif()
     set("${out_var}" "${additional_cmake_args}" PARENT_SCOPE)
 endfunction()
+
+macro(z_vcpkg_find_fortran_msys_declare_packages)
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_find_fortran.cmake:68]: Should this be a function rather than a macro?

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 2]
```diff
@@ -166,6 +166,9 @@ function(vcpkg_acquire_msys out_msys_root)
     endif()
 
     z_vcpkg_acquire_msys_declare_all_packages()
+    if(NOT "${arg_Z_DECLARE_EXTRA_PACKAGES_COMMAND}" STREQUAL "")
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_acquire_msys.cmake:169]:
```suggestion
    if(DEFINED arg_Z_DECLARE_EXTRA_PACKAGES_COMMAND)
```

**Review by BillyONeal — APPROVED** [review]
It isn't worth resetting build for my nitpicks

File: scripts/cmake/vcpkg_find_fortran.cmake (hunk 1 above)
Comment (dg0yt) [reply scripts/cmake/vcpkg_find_fortran.cmake:68]: It is the same as `z_vcpkg_acquire_msys_declare_all_packages`.
And these callables are working via side effects (although on cache variables), not as pure functions,

File: scripts/cmake/vcpkg_acquire_msys.cmake (hunk 2 above)
Comment (dg0yt) [reply scripts/cmake/vcpkg_acquire_msys.cmake:169]: I dont want to `cmake_language(CALL "")`.

Comment (BillyONeal) [reply scripts/cmake/vcpkg_acquire_msys.cmake:169]: > I dont want to `cmake_language(CALL "")`.

I feel like if someone did `Z_DECLARE_EXTRA_PACKAGES_COMMAND ""` that's kinda what they asked for. But again not worth a reset over.

**Pull request merged**